enterprise/edx_api.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+"""
+Utilities to get details from the course catalog API.
+"""
+from __future__ import absolute_import, unicode_literals
+
+import requests
+from edx_rest_api_client.client import EdxRestApiClient
+
+from django.conf import settings
+
+
+class EnrollmentApiClient(object):
+    """
+    Object builds an API client to make calls to the Enrollment API.
+    """
+
+    def __init__(self):
+        """
+        Create an Enrollment API client, authenticated with the API token from Django settings.
+        """
+        session = requests.Session()
+        session.headers = {"X-Edx-Api-Key": settings.EDX_API_KEY}
+        self.client = EdxRestApiClient(
+            settings.ENTERPRISE_ENROLLMENT_API_URL, append_slash=False, session=session
+        )
+
+    def get_course_details(self, course_id):
+        """
+        Query the Enrollment API for the course details of the given course_id.
+        """
+        return self.client.course(course_id).get()
+
+    def enroll_user_in_course(self, username, course_id, mode):
+        """
+        Query the enrollment API to enroll the user in the course specified by course_details.
```

Comment:
I'll make those changes; these were largely moved directly from the current position without modification.